TACC/tup-ui
# feat(core-styles): extend root postcss config

## Overview:

In Core-Styles extend root PostCSS config file, so **all** apps use the same (base) PostCSS config.

<details open><summary>To Do</summary>

- [x] self-review again (been a while)
- [x] manually test again
- [x] fix so manual test steps pass
- [x] re-evaluate the purpose and any problems
- [ ] make core-styles dir not be source of for truth for PostCSS dependencies and config﹡

<sub>﹡ I can’t have core-styles rely _directly_ on config nor dependencies in root, because everything in root is outside of the core-styles NPM package publish _and_ the CMS needs those in the core-styles NPM package (because CMS uses core-styles build process to parse some of its source files).</sub>

</details>

## Related:

- builds on https://github.com/TACC/tup-ui/pull/14 (specifically, 4f4571a)

## Changes:

- remove from core-styles base postcss config base what is now in root postcss config
- when extended postcss config, build off of root

## Testing:

<details><summary>1. Confirm that local core-styles build does <strong>not</strong> produce different diff.</summary>

0. Reset any code changes from previous testing.
1. Open base branch.
2. Run (`npx `)`nx build core-styles --skip-nx-cache`.
3. Open this branch.
4. Comment out certain code:
    - [core-styles' git ignore of dist](https://github.com/TACC/tup-ui/blob/082dbbb/libs/core-styles/.gitignore#L5) (so we can stage `build/`)
5. Stage the changes to `libs/core-styles/dist/`.
6. Run (`npx `)`nx build core-styles --skip-nx-cache`.
7. Confirm there *are __not__* un-staged changes.

</details>
<details><summary>2. Verify that the changes are effectual (local core-styles build—without root config—<strong>does</strong> produce different diff).</summary>

0. Reset any code changes from previous testing.
1. Open base branch.
2. Run (`npx `)`nx build core-styles --skip-nx-cache`.
3. Open this branch.
4. Comment out certain code:
    - [core-styles' git ignore of dist](https://github.com/TACC/tup-ui/blob/082dbbb/libs/core-styles/.gitignore#L5) (so we can stage `build/`)
    - [core-styles' use of root postcss config](https://github.com/TACC/tup-ui/blob/f4948b6/libs/core-styles/src/bin/config.js#L24) (so we do **not** use base config)
5. Stage the changes to `libs/core-styles/dist/`.
6. Run (`npx `)`nx build core-styles --skip-nx-cache`.
7. Confirm there *__are__* un-staged changes (because we are missing some config).

</details>

## UI:

N/A
